app/controllers/users/add_user.js

```diff
@@ -0,0 +1,148 @@
+
+  titles: Ember.computed(function() {
+    let translation = this.get("i18n");
+    let mr = translation.t("order.user_title.mr");
+    let mrs = translation.t("order.user_title.mrs");
+    let miss = translation.t("order.user_title.miss");
+    let ms = translation.t("order.user_title.ms");
+
+    return [
+      { name: mr, id: "Mr" },
+      { name: mrs, id: "Mrs" },
+      { name: miss, id: "Miss" },
+      { name: ms, id: "Ms" }
+    ];
+  }),
+
+  setEmailorMobile: Ember.computed("email", "mobilePhone", function() {
+    const email = this.get("email");
+    const mobile = this.get("mobilePhone");
+
+    if (/^[456789]\d{7}/.test(mobile) || /^[^@\s]+@[^@\s]+/.test(email)) {
+      return false;
+    } else {
+      return true;
+    }
+  }),
+
+  districts: Ember.computed(function() {
+    return this.get("store")
+      .peekAll("district")
+      .sortBy("name");
+  }),
+
+  languages: Ember.computed(function() {
+    let translation = this.get("i18n");
+    let English = translation.t("users.languages.english");
+    let Chinese = translation.t("users.languages.chinese");
+
+    return [{ name: English, id: "en" }, { name: Chinese, id: "zh-tw" }];
+  }),
+
+  locale: function(str) {
+    return this.get("i18n").t(str);
+  },
+
+  clearFormData() {
+    this.set("selectedTitle", null);
+    this.set("firstName", "");
+    this.set("lastName", "");
+    this.set("mobilePhone", "");
+    this.set("email", "");
+    this.set("selectedDistrict", null);
+    this.set("selectedLanguage", null);
+    this.set("newUploadedImage", null);
+  },
+
+  formatMobileNumber() {
+    const mobile = this.get("mobilePhone");
+    if (mobile.length) {
+      return config.APP.HK_COUNTRY_CODE + mobile;
+    }
+  },
+
+  getRequestParams() {
+    const mobilePhone = this.get("mobilePhone")
+      ? this.formatMobileNumber()
+      : "";
+    let title = this.get("selectedTitle") ? this.get("selectedTitle").id : "Mr";
+    let language = this.get("selectedLanguage")
+      ? this.get("selectedLanguage").id
+      : null;
+    let district = this.get("selectedDistrict")
+      ? this.get("selectedDistrict").id
+      : null;
+    var params = {
+      title: title,
+      first_name: this.get("firstName"),
+      last_name: this.get("lastName"),
+      mobile: mobilePhone,
+      email: this.get("email"),
+      preferred_language: language,
+      address_attributes: {
+        district_id: district,
+        addressable_type: "profile"
+      }
+    };
+    return { user: params };
+  },
+
+  saveImage() {
+    let image = this.get("newUploadedImage");
+    if (image) {
+      return image.save();
+    }
+  },
+
+  actions: {
+    back() {
+      this.clearFormData();
+      this.transitionToRoute("manage_users");
+    },
+
+    saveUser() {
+      return this.runTask(async () => {
+        let data = await new AjaxPromise(
+          "/users",
+          "POST",
+          this.get("session.authToken"),
+          this.getRequestParams()
+        );
```

Comment:
the part where a request is fired, can be put in a user-service

```suggestion
        let data = await this.get('userService').create(this.getRequestParams())
```
and the userService should handle the request and authToken thing